### ISSUE TITLE
Allow changing weight decay after initialization

### ISSUE THREAD

**Issue by danielhers** [clab/dynet#1093]
Currently weight decay is set once before DyNet is initialized and can't be changed during the whole runtime, even if training a new model.
It should be possible to define a new weight decay for a new model. This is essential for hyperparameter tuning, for example.
Related to #1071.
